yjit/src/backend/ir.rs

```diff
@@ -1031,6 +1031,10 @@ pub struct Assembler {
     /// Context for generating the current insn
     pub ctx: Context,
 
+    /// The current ISEQ's local table size. asm.local_opnd() uses this, and it's
+    /// sometimes hard to pass this value, e.g. asm.spill_temps() in asm.ccall().
+    pub num_locals: Option<u32>,
```

Comment:
Can you document in the comment in which cases this would be `None` ?

Reply:
After https://github.com/ruby/ruby/pull/11157#discussion_r1676236891, I checked if we can remove `Option`, but there were too many places and it didn't make sense to me to just use `0` for them as a compromise.

I prepared `Assembler::new_without_iseq` separately from `Assembler::new` for clarity https://github.com/ruby/ruby/pull/11157/commits/11ac40a3df6d044a4a06912bd147d7a03673a91d. The commit also added a comment https://github.com/ruby/ruby/pull/11157#discussion_r1676386533 that explains when it's `None`.